Replace fork by posix_spawn in mount_util.c

bsbernd opened the pull request:
This completes commit bb9cecbf67341e03e53dd36b8f69520f3a26f834
("Use posix_spawn instead of fork+exec") and replaces remaining
fork by posix_spawn.

Not modified is mount_bsd.d as I cannot test it.

Another commit adds the env FUSE_NO_MTAB_UPDATE to be able to
disable mtab updates without recompilation.

bsbernd committed:
Replace fork by posix_spawn in mount_util.c

This completes commit bb9cecbf67341e03e53dd36b8f69520f3a26f834
("Use posix_spawn instead of fork+exec") and replaces remaining
fork by posix_spawn.

Not modified is mount_bsd.d as I cannot test it.

Signed-off-by: Bernd Schubert <bernd@bsbernd.com>

trapexit commented on lib/mount_util.c:
```diff
@@ -81,13 +91,45 @@ static int mtab_needs_update(const char *mnt)
+
+	int flags = POSIX_SPAWN_RESETIDS | POSIX_SPAWN_SETSIGMASK;
+	res = posix_spawnattr_setflags(attr, flags);
+	if (res) {
```
Why the difference between this check and the one on line 110? From the man pages both appear to return 0 on success and an errno on error.

bsbernd replied:
Thanks,  the check below for (res < 0) needs to be fixed. Although I see it in line 103

bsbernd committed:
mount_util: derive add/remove-mount result from exit status

add_mount(), exec_umount() and remove_mount() returned waitpid()'s pid
on success rather than 0, contradicting the documented '0 or -1' contract
(harmless today only because every caller tests == -1). They also tested
the raw wait status, and read it uninitialized when waitpid() itself
failed.

Read the status only on the success path and decode it with
WIFEXITED()/WEXITSTATUS(), matching how check_is_mount() in fusermount.c
already does it, so these helpers return 0 or -1 as advertised.

Signed-off-by: Bernd Schubert <bernd@bsbernd.com>
